src/solvers/refinement/string_refinement.cpp

```diff
@@ -896,6 +871,44 @@ exprt string_refinementt::substitute_array_with_expr(
   }
 }
 
+/// Fill an array represented by a list of with_expr by propagating values to
+/// the left. For instance `ARRAY_OF(12) WITH[2:=24] WITH[4:=42]` will give
+/// `{ 24, 24, 24, 42, 42 }`
+/// \param expr: an array expression in the form
+///   `ARRAY_OF(x) WITH [i0:=v0] ... WITH [iN:=vN]`
+/// \param string_max_length: bound on the length of strings
+/// \return an array expression with filled in values, or expr if it is simply
+///   an `ARRAY_OF(x)` expression
+exprt fill_in_array_with_expr(const exprt &expr, std::size_t string_max_length)
```

Comment:
I've added one for `concretize_string` which should cover this function https://github.com/diffblue/cbmc/pull/1241/files#diff-da82b969a8fb1d04473386fa029dcc4fR17